reporters/reporter-default/src/main/java/org/radargun/reporting/commons/DataReporter.java

```diff
@@ -0,0 +1,175 @@
+         rowData.put(operationName + ".Requests", String.valueOf(defaultOutcome.requests));
+         rowData.put(operationName + ".Errors", String.valueOf(defaultOutcome.errors));
+         rowData.put(operationName + ".ResponseTimeMax", String.valueOf(defaultOutcome.responseTimeMax));
+         rowData.put(operationName + ".ResponseTimeMean", String.valueOf(round(defaultOutcome.responseTimeMean, 2)));
+      }
+      OperationThroughput throughput = statistics.getRepresentation(operationName, OperationThroughput.class);
+      if (throughput != null) {
+         rowData.put(operationName + ".ThroughputWithErrors", String.valueOf(round(throughput.gross, 1)));
+         rowData.put(operationName + ".Throughput", String.valueOf(round(throughput.net, 1)));
+      }
+      for (double percentile : percentiles) {
+         Percentile result = statistics.getRepresentation(operationName, Percentile.class, percentile);
+         if (result != null) {
+            rowData.put(operationName + ".RTM_" + percentile, String.valueOf(round(result.responseTimeMax, 2)));
+         }
+      }
+      MeanAndDev meanAndDev = statistics.getRepresentation(operationName, MeanAndDev.class);
+      if (meanAndDev != null) {
+         rowData.put(operationName + ".ResponseTimeMean", String.valueOf(round(meanAndDev.mean, 2)));
+         rowData.put(operationName + ".ResponseTimeDeviation", String.valueOf(round(meanAndDev.dev, 2)));
+      }
+      DataThroughput dataThroughput = statistics.getRepresentation(operationName, DataThroughput.class);
+      if (dataThroughput != null) {
+         rowData.put(operationName + ".DataThrouputMin", String.valueOf(round(dataThroughput.minThroughput, 1)));
+         rowData.put(operationName + ".DataThrouputMax", String.valueOf(round(dataThroughput.maxThroughput, 1)));
+         rowData.put(operationName + ".DataThrouputMean", String.valueOf(round(dataThroughput.meanThroughput, 1)));
+         rowData.put(operationName + ".DataThrouputStdDeviation", String.valueOf(round(dataThroughput.deviation, 2)));
+         rowData.put(operationName + ".TotalBytes", String.valueOf(dataThroughput.totalBytes));
+         rowData.put(operationName + ".ResponseTimes", Arrays.toString(dataThroughput.responseTimes).replace(",", ""));
+      }
+   }
+
+   private static double round(double number, int places) {
+      BigDecimal bigDecimal = new BigDecimal(number).setScale(places, RoundingMode.HALF_UP);
+      return bigDecimal.doubleValue();
+   }
+
+   public static File prepareOutputFile(Report report, String prefix, String suffix, String targetDir, String extension) throws IOException {
+      File parentDir = new File(targetDir);
+      if (!parentDir.exists()) {
+         if (!parentDir.mkdirs()) {
+            log.warn("Issues creating parent dir " + parentDir);
+         }
+      } else {
+         if (!parentDir.isDirectory()) {
+            throw new IllegalStateException(targetDir + " is not a directory");
+         }
+      }
```

Comment:
I think you can replace those lines with
```
try {
            Files.createDirectory(file.toPath());
        } catch (IOException e) {
            throw new IllegalStateException("Issues creating parent dir " + parentDir);
        }
 ```
 The Files.createDirectories() creates a new directory and parent directories that do not exist. This method does not throw an exception if the directory already exists.

Reply:
The content of `DataReporter` comes from `CsvReporter`. How can I add the warning and throw the exception if the `parentDir` is a file using `Files.createDirectory`?